docs/reference/rest-api/logs.asciidoc

```diff
@@ -0,0 +1,278 @@
+
+TIP: Use the {ecs-logging-ref}/intro.html[ECS logging libraries] to create ECS-compliant JSON logs.
+
+Only the following fields, which are commonly used for searching, filtering, and correlating logs, are indexed by default:
+
+* {ecs-ref}/ecs-base.html#field-timestamp[`@timestamp`] +
+  If not provided, will be set to the current time.
+  Can be provided as <<epoch-millis,`epoch_millis`>> or <<strict-date-time,`strict_date_optional_time`>> by default.
+* {ecs-ref}/ecs-data_stream.html#field-data-stream-dataset[`data_stream.dataset`] +
+  Overrides the `data_stream.dataset` <<logs-api-path-params, path parameter>> on a per-event basis.
+* {ecs-ref}/ecs-data_stream.html#field-data-stream-namespace[`data_stream.namespace`] +
+  Overrides the `data_stream.namespace` <<logs-api-path-params, path parameter>> on a per-event basis.
+* {ecs-ref}/ecs-base.html#field-message[`message`]
+* {ecs-ref}/ecs-log.html#field-log-level[`log.level`]
+* {ecs-ref}/ecs-log.html#field-log-logger[`log.logger`]
+* {ecs-ref}/ecs-service.html#field-service-name[`service.name`]
+* {ecs-ref}/ecs-service.html#field-service-environment[`service.environment`]
+* {ecs-ref}/ecs-service.html#field-service-version[`service.version`]
+* {ecs-ref}/ecs-tracing.html#field-trace-id[`trace.id`]
+* {ecs-ref}/ecs-tracing.html#field-transaction-id[`transaction.id`]
+* {ecs-ref}/ecs-tracing.html#field-span-id[`span.id`]
+* {ecs-ref}/ecs-process.html#field-process-pid[`process.pid`]
+* {ecs-ref}/ecs-process.html#field-process-thread-name[`process.thread.name`]
+* {ecs-ref}/ecs-error.html#field-error-type[`error.type`]
+* {ecs-ref}/ecs-error.html#field-error-message[`error.message`]
+* {ecs-ref}/ecs-event.html#field-event-dataset[`event.dataset`]
+* {ecs-ref}/ecs-cloud.html#field-cloud-provider[`cloud.provider`]
+* {ecs-ref}/ecs-cloud.html#field-cloud-availability-zone[`cloud.availability_zone`]
+* {ecs-ref}/ecs-cloud.html#field-cloud-region[`cloud.region`]
+* {ecs-ref}/ecs-host.html#field-host-hostname[`host.hostname`]
+* {ecs-ref}/ecs-host.html#field-host-name[`host.name`]
+* {ecs-ref}/ecs-container.html#field-container-id[`container.id`]
+* {ecs-ref}/ecs-container.html#field-container-name[`container.name`]
+* {ecs-ref}/ecs-orchestrator.html#field-orchestrator-namespace[`orchestrator.namespace`]
+* {ecs-ref}/ecs-orchestrator.html#field-orchestrator-cluster-id[`orchestrator.cluster.id`]
+* {ecs-ref}/ecs-orchestrator.html#field-orchestrator-cluster-name[`orchestrator.cluster.name`]
+* {ecs-ref}/ecs-orchestrator.html#field-orchestrator-resource-id[`orchestrator.resource.id`]
+* {ecs-ref}/ecs-orchestrator.html#field-orchestrator-resource-name[`orchestrator.resource.name`]
+
+Dotted field names are expanded to objects so that they can be used interchangeably with nested objects. For example, the following documents are treated equally: `{"log.level": "INFO"}`, `{"log": { "level": "INFO"} }`.
```

Comment:
Should we directly start to use subobject false to prevent conflicts?

Reply:
I think we should but there are some missing features. See: https://github.com/elastic/elasticsearch/issues/88934

Reply:
Another reason for expanding dots is to be able to access the fields in a consistent way in ingest pipelines and scripts so that you don't need to have something like `_doc['log']['level']` and `_doc['log.level']`.

Reply:
That argument can be taken both ways. I think the important bit is that it is consistent, either all flat or all expanded so that we don't have something like `_doc['foo']['hello.world']`